### ISSUE TITLE
[Memory-layout infra] Create explicit channel-last ops

### ISSUE THREAD

**Issue by AdrianLundell** [pytorch/executorch#20093]
Part of #19299. Create a new operator dialect `exir_ops.edge.channels_last` which
work on channels-last memory format with contiguous strides, i.e. the shape is
interpreted as (N, H, W, C), and does not change dim-order. 

The implementation includes fake kernels, aot-kernels, runtime-kernels, and testing. 

Efficiency is no requirement, so the regular ops + permutes may be used for example.

Operators to implement:
- exir_ops.edge.aten.avg_pool2d.default
- exir_ops.edge.aten.adaptive_avg_pool2d.default
- exir_ops.edge.aten.upsample_bilinear2d.vec
- exir_ops.edge.aten.upsample_nearest2d.vec
- exir_ops.edge.aten.convolution.default
- exir_ops.edge.aten.max_pool2d_with_indices.default
- exir_ops.edge.aten.max_pool2d.default (?)
- exir_ops.edge.aten.permute_copy.default
- exir_ops.edge.aten.t_copy.default (?)
- exir_ops.edge.aten.transpose_copy.int (?)

grid_sample may need to be handled as well, it is a bit special since the input is channels-first but the grid is channels-last: https://docs.pytorch.org/docs/2.12/generated/torch.nn.functional.grid_sample.html

We might want to only implement one permute operator and normalize all variants to use that. Similarly it may be enough to only implement max_pool2d_with_indices, but that removes the possibility to use RemoveGetItemPass, which will need to be handled another way in that case.

**Comment by rascani** [comment]
@AdrianLundell - I think this is functionally done now. I didn't add any runtime kernels, but I don't think we really need them, as these should really just be transient IR anyways. I'm going to close this, but feel free to re-open.

**Comment by AdrianLundell** [comment]
Yes I agree, good work
